### ISSUE TITLE
`authenticateMiddleware` function should not set user to request

### ISSUE THREAD

**Issue by binle** [lukeautry/tsoa#828]
Currently, the generated routes file exist function `authenticateMiddleware` for security section with success handler:
```
const succeed = function(user: any) {
    if (!success) {
        success = true;
        responded++;
        request['user'] = user;
        next();
    }
} 
Promise.all(promises)
                        .then((users) => { succeed(users[0]); })
                        .catch(fail);
......


```

- **I have some logic which handle request and set user to request before call other handlers in `RegisterRoutes`**, and so that, `authenticateMiddleware` overwrote the user in request.

  - [ x] bug report
  - [x ] feature request
  - [ x] support request

## Expected Behavior: don't set user to request, and let developer set user to request in `expressAuthentication` function

```
const succeed = function() {
    if (!success) {
        success = true;
        responded++;
        next();
    }
} 

....
Promise.all(promises).then(succeed).catch(fail);

```

==> if developer want set user to request, they can set it when check the authentications:
```
export async function expressAuthentication(
    request: ExRequest,
    securityName: string,
    scopes?: string[]
): Promise<unknown> {
    if (securityName === 'api_key') {
        // TODO
        request.user = {name: 'Test', id: '01'};
    }
}


```

**Comment by jackey8616** [comment]
If you want to prevent original `expressAuthentication` function, you can make your own middle like [here](https://tsoa-community.github.io/docs/authentication.html).  
  
TL;DR,  
1) defined route of authenticationModule in `tsoa.json`:  
```js
{
  ...
  "routes": {
    "authenticationModule": "./[path to your authentication.ts]",
  }
  ...
}
```
2) Make your own function:  
```js
export function expressAuthentication(
  request: express.Request,
  securityName: string,
  scopes?: string[]
): Promise<any> {
  ... Your logic ...
}
```

**Comment by binle** [comment]
@jackey8616 
I know how to use `authenticationModule` module and I known how to overwrite the routes template and generate `routes.ts` file.
I mean the default logic of template file is not general, we should make the change in file `https://github.com/lukeautry/tsoa/blob/master/packages/cli/src/routeGeneration/templates/express.hbs`
and update document for it.
It will be more flexible and developer can clear more about `authenticationModule` module.
Thanks!

**Comment by jackey8616** [comment]
@binle 
I've trace the original proposal issue for `authenticationModule` which is #20 and implemented in #58 .  
When it first comes out, there is already returns with user-like values.  

So I think with some little bit redesign for a general usage is a good idea.  
What's your opinion? @WoH
